gyp/pylib/gyp/common.py

```diff
@@ -337,11 +338,16 @@ def WriteOnDiff(filename):
   class Writer(object):
     """Wrapper around file which only covers the target if it differs."""
     def __init__(self):
+      # On Cygwin remove the "dir" argument because `C:` prefixed paths are treated as relative,
+      # consequently ending up with current dir "/cygdrive/c/..." being prefixed to those, which was
+      # obviously a non-existent path, for example: "/cygdrive/c/<some folder>/C:\<my win style abs path>".
+      # See https://docs.python.org/2/library/tempfile.html#tempfile.mkstemp for more details
+      baseTempDir = "" if IsCygwin() else os.path.split(filename)[0];
```

Comment:
Python uses snake_case (base_temp_dir), not camelCase (baseTempDir).  ;-)

Reply:
@cclauss I had left the original code intact by using `os.path.split()[0]`, but I went ahead and updated it to use `os.path.dirname()`, gave it a quick test locally, and then pushed the changed.